src/game/interpreter.rs

```diff
@@ -846,12 +847,73 @@ impl PostFlopGame {
         ret
     }
 
+    // TODO use in .strategy()
+    pub fn strategy_at_node(&self, node: &PostFlopNode) -> Vec<f32> {
+        if self.state < State::MemoryAllocated {
+            panic!("Memory is not allocated");
+        }
+
+        let player = self.current_player();
+        let num_actions = node.num_actions();
+        let num_hands = self.num_private_hands(player);
+
+        let mut ret = if self.is_compression_enabled {
+            normalized_strategy_compressed(node.strategy_compressed(), num_actions)
+        } else {
+            normalized_strategy(node.strategy(), num_actions)
+        };
+
+        let locking = self.locking_strategy(node);
+        apply_locking_strategy(&mut ret, locking);
+
+        ret.chunks_exact_mut(num_hands).for_each(|chunk| {
+            self.apply_swap(chunk, player, false);
+        });
+
+        ret
+    }
+
     /// Returns the total bet amount of each player (OOP, IP).
     #[inline]
     pub fn total_bet_amount(&self) -> [i32; 2] {
         self.total_bet_amount
     }
 
+    /// Locks the strategy of the current node to the current strategy already in memory.
+    pub fn lock_current_strategy(&mut self) {
+        if self.is_terminal_node() {
+            panic!("Terminal node is not allowed");
+        }
+
+        if self.is_chance_node() {
+            panic!("Chance node is not allowed");
+        }
+
+        let mut node = self.node();
+
+        node.is_locked = true;
```

Comment:
Small thing, but I think moving this after the `let` binding on the next line will help show that `node.is_locked == true` is tightly coupled with `self.locking_strategy` being populated at this node's index.